config/core/configmaps/deployment.yaml

```diff
@@ -76,6 +76,10 @@ data:
     # If omitted, no value is specified and the system default is used.
     queue-sidecar-ephemeral-storage-limit: "1024Mi"
 
+    # Sets tokens associated with specific audiences for the queue proxy's QPOptions
+    # If omitted, or empty, no tokens are created
+    # Supports a list of audiences e.g. queue-sidecar-tokens: "some-audience,some-other-audience"
+
```

Comment:
We should include a specific, uncommented example (similar to L77 above)

Reply:
It seems that the examples expected are the default used -  
Using queue-sidecar-tokens: "" is not the same as not adding this config at all - so we seem to be unable to include a specific, uncommented example

adding queue-sidecar-tokens: "" and running tests results in:
```
--- FAIL: TestControllerConfigurationFromFile (0.00s)
    /Users/davidhadasmac16/Code/Go/github.com/davidhadas/serving/pkg/deployment/config_test.go:72: Example stanza does not match default, diff(-want,+got):   &deployment.Config{
          	... // 8 identical fields
          	QueueSidecarEphemeralStorageRequest: nil,
          	QueueSidecarEphemeralStorageLimit:   nil,
        - 	QueueSidecarTokens:                  nil,
        + 	QueueSidecarTokens:                  sets.String{"": {}},
          	ConcurrencyStateEndpoint:            "",
          }
        
--- FAIL: TestControllerConfiguration (0.00s)
    --- FAIL: TestControllerConfiguration/legacy_keys_supported (0.00s)
        /Users/davidhadasmac16/Code/Go/github.com/davidhadas/serving/pkg/deployment/config_test.go:305: Config mismatch, diff(-want,+got):   &deployment.Config{
              	... // 8 identical fields
              	QueueSidecarEphemeralStorageRequest: &{i: {...}, s: "9M", Format: "DecimalSI"},
              	QueueSidecarEphemeralStorageLimit:   &{i: {...}, s: "10M", Format: "DecimalSI"},
            - 	QueueSidecarTokens:                  sets.String{"bar": {}},
            + 	QueueSidecarTokens:                  nil,
              	ConcurrencyStateEndpoint:            "11",
              }
```

Reply:
In that case, should we consider changing how we represent the default (nil) case, esp. since `nil` isn't a valid value for strings?

Reply:
We don't want to require people to have a QueueSidecarTokens in their config file - so a missing QueueSidecarTokens need to be supported. Maybe we can replace a missing QueueSidecarTokens with an empty one in code.

Reply:
sets.String{"": {}}, is what we get when the configuration is `queue-sidecar-tokens: "" `
This means we get a set with 1 value and the value "".  
This is just the way sets work. 
As a result, if we decide to force the default to be "" - we will always get a set with one item. 

As a result we will have:
```golang
for aud := range cfg.Deployment.QueueSidecarTokens {
	// add token for audience <aud> under filename <aud>
	addToken(tokenVolume, aud, aud, ptr.Int64(3600))
}
```	
adding a token such as: `addToken(tokenVolume,"", "", ptr.Int64(3600))`
 And to avoid adding this token we need to add protection inside addToken to not add empty string audience and filename.

Reply:
Sending a commit with these changes